Custom operators and builders (#33355)
idaholab/moose: framework/doc/content/source/mfem/actions/AddMFEMProblemComposerAction.md

# AddMFEMProblemComposerAction

!if! function=hasCapability('mfem')

## Overview

Action called to add a `MFEMProblemComposer` that builds `ProblemOperator`s in the executioner.

> [`MFEMProblemComposer`](syntax/MFEMProblemComposer/index.md) block in the user input.

Reviewer: ```suggestion
Action called to add a problem composer, an object that builds [ProblemOperator.md]s,
parsing content inside a [`ProblemComposer`](syntax/ProblemComposer/index.md) block in the user input.
```

(I'm asking for a block name change)

Author: Done pre-squash

Only has an effect if the `Problem` type is set to

> [MFEMProblem.md](source/mfem/problem/MFEMProblem.md).

Reviewer: ```suggestion
Only has an effect if the `Problem` type is set to [MFEMProblem.md].
```

Author: applied pre-squash

## Example Input File Syntax

!listing test/tests/mfem/problemcomposers/prob_op_block_darcy.i block=Problem MFEMProblemComposer

> !listing test/tests/mfem/problemcomposers/prob_op_mfem_multiple_timesequences.i block=Problem MFEMProblemComposer

Reviewer: ```suggestion
!listing test/tests/mfem/problemcomposers/prob_op_block_darcy.i block=Problem ProblemComposer

!listing test/tests/mfem/problemcomposers/prob_op_mfem_multiple_timesequences.i block=Problem ProblemComposer
```

I think we might need the extra space to make both examples show on the website.

Reviewer: The extra space did solve that issue, but we now need to update the syntax ref to reflect the new block name.

# AddMFEMProblemComposerAction

!if! function=hasCapability('mfem')

## Overview

Action called to add a problem composer, an object that builds [ProblemOperator.md]s,

> parsing content inside a [`ProblemComposer`](syntax/ProblemComposer/index.md) block in the user

Reviewer: Let's pluralise the syntax block name to `ProblemComposers`, just so we're future-proofed on that front. We won't do anything else to allow multiple composers for now, but it's just so we don't have to change syntax down the line.

Author: I have done this, and changed it in all dependant places in documentation

Author: I believe I have done that
